packages/x-components/src/x-modules/next-queries/components/next-queries-list.vue

```diff
@@ -0,0 +1,278 @@
+<template>
+  <NoElement>
+    <!--
+      @slot Next queries list layout.
+        @binding {SearchItem[]} items - Next queries groups plus the injected search items to
+        render.
+        @binding {Vue | string} animation - Animation to animate the elements.
+    -->
+    <slot v-bind="{ items, animation }">
+      <ItemsList :animation="animation" :items="items">
+        <template v-for="(_, slotName) in $scopedSlots" v-slot:[slotName]="{ item }">
+          <slot :name="slotName" :item="item" />
+        </template>
+      </ItemsList>
+    </slot>
+  </NoElement>
+</template>
+
+<script lang="ts">
+  import { NextQuery } from '@empathyco/x-types';
+  import { mixins } from 'vue-class-component';
+  import { Component, Prop } from 'vue-property-decorator';
+  import { Getter } from '../../../components/decorators/store.decorators';
+  import { NoElement } from '../../../components/no-element';
+  import { ItemsListInjectionMixin } from '../../../components/items-list-injection.mixin';
+  import ItemsList from '../../../components/items-list.vue';
+  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { groupItemsBy, ListItem } from '../../../utils';
+  import ResultsList from '../../search/components/results-list.vue';
+  import { nextQueriesXModule } from '../x-module';
+
+  /**
+   * A list of next queries.
+   */
+  interface NextQueriesGroup extends ListItem {
+    modelName: 'NextQueriesGroup';
+    nextQueries: NextQuery[];
+  }
+
+  /**
+   * Component that inserts groups of next queries in different positions of the injected search
+   * items list, based on the provided configuration.
+   *
+   * @public
+   */
+  @Component({
+    components: {
+      ResultsList,
+      NoElement,
+      ItemsList
+    },
+    mixins: [xComponentMixin(nextQueriesXModule)]
+  })
+  export default class NextQueriesList extends mixins(ItemsListInjectionMixin) {
+    /**
+     * Animation component that will be used to animate the next queries groups.
+     *
+     * @public
+     */
+    @Prop()
+    protected animation?: Vue | string;
+
+    /**
+     * The first index to insert a group of next queries at.
+     *
+     * @public
+     */
+    @Prop({ default: 24 })
+    public offset!: number;
+
+    /**
+     * The items cycle size to keep inserting next queries groups at.
+     *
+     * @public
+     */
+    @Prop({ default: 24 })
+    public frequency!: number;
+
+    /**
+     * The maximum amount of next queries to add in a single group.
+     *
+     * @public
+     */
+    @Prop({ default: 4 })
+    public maxNextQueriesPerGroup!: number;
```

Comment:
`maxItemsToRender` to keep naming consistency?

Reply:
I think that's  confusing for this prop, here you are talking about a list of lists, and `maxItemsToRender` does not explicitly say if you are talking about the amount of groups or about the amount of next queries to render in each group.